enshrine-esdt-safe/tests/enshrine_esdt_safe_blackbox_test.rs

```diff
@@ -441,3 +585,264 @@ fn test_register_tokens_insufficient_wegld() {
         }),
     );
 }
+
+#[test]
+fn test_deposit_nothing_to_transfer() {
+    let mut state = EnshrineTestState::new();
+    let amount = BigUint::from(10000u64);
+    let wegld_payment = EsdtTokenPayment::new(WEGLD_IDENTIFIER.into(), 0, amount.clone());
+    let mut payments = PaymentsVec::new();
+    let error_status = ErrorStatus {
+        code: 4,
+        error_message: "Nothing to transfer",
+    };
+
+    payments.push(wegld_payment);
+
+    state.propose_setup_contracts(false, true, None, None);
+    state.propose_deposit(
+        ENSHRINE_ESDT_OWNER_ADDRESS,
+        USER_ADDRESS,
+        payments,
+        OptionalValue::None,
+        Some(error_status),
+    );
+}
+
+#[test]
+fn test_deposit_max_transfers_exceeded() {
+    let mut state = EnshrineTestState::new();
+    let amount = BigUint::from(10000u64);
+    let wegld_payment = EsdtTokenPayment::new(WEGLD_IDENTIFIER.into(), 0, amount.clone());
+    let mut payments = PaymentsVec::new();
+    let error_status = ErrorStatus {
+        code: 4,
+        error_message: "Too many tokens",
+    };
+
+    payments.extend(std::iter::repeat(wegld_payment).take(11));
+
+    state.propose_setup_contracts(false, false, None, None);
+    state.propose_deposit(
+        ENSHRINE_ESDT_OWNER_ADDRESS,
+        USER_ADDRESS,
+        payments,
+        OptionalValue::None,
+        Some(error_status),
+    );
+}
+
+#[test]
+fn test_deposit_no_transfer_data() {
+    let mut state = EnshrineTestState::new();
+    let amount = BigUint::from(10000u64);
+    let wegld_payment = EsdtTokenPayment::new(WEGLD_IDENTIFIER.into(), 0, amount.clone());
+    let fungible_payment = EsdtTokenPayment::new(FUNGIBLE_TOKEN_ID.into(), 0, amount.clone());
+    let crowd_payment = EsdtTokenPayment::new(CROWD_TOKEN_ID.into(), 0, amount.clone());
+    let mut payments = PaymentsVec::new();
+    let mut tokens_whitelist = MultiValueEncoded::new();
+    tokens_whitelist.push(WEGLD_IDENTIFIER.into());
+    tokens_whitelist.push(CROWD_TOKEN_ID.into());
+
+    payments.push(wegld_payment);
+    payments.push(fungible_payment);
```

Comment:
Because FUNGIBLE_TOKEN_ID is not whitelisted, it should be refunded. Please check that fungible token amount == `amount` at line 682